public/app/features/dashboard/components/DashNav/DashNav.tsx

```diff
@@ -321,6 +322,10 @@ export const DashNav = React.memo<Props>((props) => {
       return [renderPlaylistControls(), renderTimeControls()];
     }
 
+    if (props.isPublic && !!props.dashboard.meta.publicDashboardTimeSelectionEnabled) {
+      return [renderTimeControls()];
+    }
```

Comment:
renderTimeControls already has a conditional that comes from `dashboard.timepicker.hidden`, public dashboards should find a way to set this to true when it want to disable / hide the time picker